server/v2/stf/stf.go

```diff
@@ -300,9 +306,13 @@ func (s STF[T]) validatorUpdates(ctx context.Context, state store.GetWriter) ([]
 }
 
 // Simulate simulates the execution of a tx on the provided state.
-func (s STF[T]) Simulate(ctx context.Context, state store.GetReader, gasLimit uint64, tx T) (appmanager.TxResult, store.GetWriter) {
+func (s STF[T]) Simulate(ctx context.Context, state store.GetReader, gasLimit uint64, tx T) (appmanager.TxResult, []store.StateChanges) {
 	simulationState := s.branch(state)
-	return s.deliverTx(ctx, simulationState, tx), simulationState
+	cs, err := simulationState.GetStateChanges()
+	if err != nil {
+		return appmanager.TxResult{}, nil
+	}
+	return s.deliverTx(ctx, simulationState, tx, corecontext.ExecModeSimulate), cs
```

Comment:
I think these two calls should be inverted, first deliverTx and then GetStateChanges

Reply:
switched

Reply:
that was fast